src/App.jsx

```diff
@@ -14,45 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_BY_LENGTH = 'by length';
+const FALSE = false;
+
+function getPreparedGoods(goods, { sortField, reverse }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const readyGoods = getPreparedGoods(goodsFromServer, { sortField, reverse });
+
+  const doSortAlphabetically = () => setSortField(SORT_FIELD_ALPHABETICALLY);
+  const doSortByLength = () => setSortField(SORT_FIELD_BY_LENGTH);
+  const doReverse = () => setReverse(!reverse);
+  const doReset = () => {
+    setSortField('');
+    setReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn('button', 'is-info',
+              { 'is-light': sortField !== SORT_FIELD_ALPHABETICALLY })}
```

Comment:
```suggestion
            cn(
              'button',
              'is-info',
              { 'is-light': sortField !== SORT_FIELD_ALPHABETICALLY }
            )}
```
Pay attention to formatting